firestore/query.go

```diff
@@ -687,6 +695,15 @@ func (f filter) toProto() (*pb.StructuredQuery_Filter, error) {
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := arrayOperators[op]; ok {
+		switch val.GetValueType().(type) {
+		case *pb.Value_ArrayValue:
+			break
+		default:
+			log.Printf("WARNING firestore: operator %q requires an array value", f.op)
```

Comment:
I'm not sure about this error message. I agree with the sentiment--let's tell developers that there's an issue here--but I'm not sure whether an all-caps "WARNING" is the best way to go.

@codyoss WDYT?

Reply:
In general we have refrained from any logging in our client libraries. Without proper wholistic support there is not a great way for users to turn off logs if something like this is done except with some global flags that could affect there code as well.

Reply:
Given we already have the issue logged and scheduled for fix in the next major breaking change https://github.com/googleapis/google-cloud-go/issues/6330, I think that not logging this warning is also fine. So it seems we can abandon this PR? 

What other information or stakeholders do we need to make a decision here?